### PR TITLE
SCP-2510: allow making zero input transactions

### PR DESCRIPTION
**Pull request by sjoerdvisscher**
Pre-submit checklist:
- Branch
    - [ ] Tests are provided (if possible)
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Relevant tickets are mentioned in commit messages
    - [x] Formatting, materialized Nix files, PNG optimization, etc. are updated
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [x] Reviewer requested

### CODE REVIEW

File: plutus-contract/src/Plutus/Contract/CardanoAPITemp.hs [hunk 1]
```diff
@@ -0,0 +1,161 @@
+{-# LANGUAGE BlockArguments           #-}
+{-# LANGUAGE DisambiguateRecordFields #-}
+{-# LANGUAGE EmptyCase                #-}
+{-# LANGUAGE FlexibleContexts         #-}
+{-# LANGUAGE GADTs                    #-}
+{-# LANGUAGE NamedFieldPuns           #-}
+{-# LANGUAGE TypeApplications         #-}
+-- Code temporarily copied over from cardano-api.
```
Comment (michaelpj) [inline plutus-contract/src/Plutus/Contract/CardanoAPITemp.hs:8]: why? when will we get rid of it? better out of your head!

Comment (sjoerdvisscher) [reply plutus-contract/src/Plutus/Contract/CardanoAPITemp.hs:8]: Well, it's not in my head but in Jann's 😅, but I'll try to add a clarification.

File: plutus-contract/src/Plutus/Contract/CardanoAPITemp.hs [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+        txExtraKeyWits,
+        txWithdrawals,
+        txCertificates,
+        txMintValue,
+        txScriptValidity
+    } =
+    return $
+      ShelleyTxBody ShelleyBasedEraAlonzo
+        (Alonzo.TxBody
+          (Set.fromList (map (toShelleyTxIn . fst) txIns))
```
Comment (raduom) [inline plutus-contract/src/Plutus/Contract/CardanoAPITemp.hs:54]: As far as changes from the original, you just removed the guard here?

**Review by raduom — APPROVED** [review]
LGTM

File: plutus-contract/src/Plutus/Contract/CardanoAPITemp.hs (hunk 2 above)
Comment (sjoerdvisscher) [reply plutus-contract/src/Plutus/Contract/CardanoAPITemp.hs:54]: Yes.

**Pull request merged**